### DISCUSSION TITLE
DynamicVector in for in loop produces references to elements instead of values

### DISCUSSION THREAD

**Discussion by jolonf** [modular/modular#2013] [Q&A] [answered]
When I write a for in loop using a DynamicVector, when I use the loop variable representing the element, it generates an error similar to:

`(error: 'Reference[SIMD[si8, 1], 1, dv]' value has no attribute 'to_int')`

The loop variable is a reference rather than the value. Is this correct? How can I get access to the value rather than the reference?

```mojo
var dv = DynamicVector[Int8]()
    
dv.append(1)
dv.append(2)
    
for b in dv:
  print(b.to_int())
```

Indexing the DynamicVector works fine:

```mojo
for i in range(len(dv)):
  print(dv[i].to_int())
```

**Comment by soraros** [comment, marked as answer]
You can deref by using `b[]`.
